### PR TITLE
PO-471

### PR DESCRIPTION
**Pull request by iamfrankiemoran**
### Jira link
[PO-471](https://tools.hmcts.net/jira/browse/PO-471)
[PO-649](https://tools.hmcts.net/jira/browse/PO-649)

### Change description
- Collection order now showing for Adult or youth only
- Collection order now showing for Adult or youth with parent/guardian to pay
- Make collection order today works off the collection order permission

### Testing done
- Tested locally and all working as expected
- Unit tests working as expected
- Smoke test passing locally
- Functional tests have 8 failing tests due to the new Collection Order error messaging

### Checklist

- [x] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

**Review by max-holland — APPROVED** [review]

File: cypress/e2e/functional/opal/paymentTerms/PO-471_collectionOrder_adultOrYouth.feature [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+        And I click on the "Payment terms" link
+        Then I see "Payment terms" on the page header
+
+        And I select the "No" radio button under the "Has a collection order been made?" section
+        And I select the "Make collection order today" checkbox
+        When I select the "Pay in full" radio button
+        And I see "For example, 31/01/2023" hint text above the "Enter pay by date" date picker
+        And I enter "12/09/2023" into the "Enter pay by date" date field
+
+    Scenario Outline: AC4 -negative: If the user has not selected a value for the 'Has a collection order been made?' field and clicks either the 'Return to account details' or 'Add account comments and notes' buttons
```
Comment (CadeFaulkner) [inline cypress/e2e/functional/opal/paymentTerms/PO-471_collectionOrder_adultOrYouth.feature:55]: not an issue with the test at all, and I know this isnt ideal but due to the time constraints we have in the pipelines, would you be able to combine these into a single scenario, I believe it should be fairly straightforward.

File: cypress/e2e/functional/opal/paymentTerms/PO-471_collectionOrder_adultOrYouth.feature [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+        And I select the "Yes" radio button under the "Has a collection order been made?" section
+        Then I see "Payment terms" on the page header
+        When I click the "<returnButton>" button
+        Then I see the error message "Enter date collection order made" at the top of the page
+
+        Examples:
+            | returnButton                   |
+            | Return to account details      |
+            | Add account comments and notes |
```
Comment (CadeFaulkner) [inline cypress/e2e/functional/opal/paymentTerms/PO-471_collectionOrder_adultOrYouth.feature:81]: same comment as above

**Review by CadeFaulkner — CHANGES_REQUESTED** [review]

File: cypress/e2e/functional/opal/paymentTerms/PO-471_collectionOrder_adultOrYouth.feature (hunk 1 above)
Comment (rajanigandra) [reply cypress/e2e/functional/opal/paymentTerms/PO-471_collectionOrder_adultOrYouth.feature:55]: I haven't created these 2 stpes. re-using from other feature files

File: cypress/e2e/functional/opal/paymentTerms/PO-471_collectionOrder_adultOrYouth.feature (hunk 2 above)
Comment (rajanigandra) [reply cypress/e2e/functional/opal/paymentTerms/PO-471_collectionOrder_adultOrYouth.feature:81]: No Problem at all. Same steps I could repeat in the same scenario. In this way we could avoid 'Background' steps twice running

File: cypress/e2e/functional/opal/paymentTerms/PO-471_collectionOrder_adultOrYouth.feature (hunk 1 above)
Comment (rajanigandra) [reply cypress/e2e/functional/opal/paymentTerms/PO-471_collectionOrder_adultOrYouth.feature:55]: I kept these 2 steps in one scenarios and on the rest of the scenarios we don't necessarily need so removed

Comment (rajanigandra) [reply cypress/e2e/functional/opal/paymentTerms/PO-471_collectionOrder_adultOrYouth.feature:55]: In one step we are verifying the text presented on the screen, On the other step entering the value on to a specific field

**Review by CadeFaulkner — APPROVED** [review]

**Pull request merged**